src/services/codefixes/convertToTypeOnlyExport.ts

```diff
@@ -44,13 +44,17 @@ namespace ts.codefix {
                 exportDeclaration.modifiers,
                 /*isTypeOnly*/ false,
                 factory.updateNamedExports(exportClause, filter(exportClause.elements, e => !contains(typeExportSpecifiers, e))),
-                exportDeclaration.moduleSpecifier);
+                exportDeclaration.moduleSpecifier,
+                /*assertClause*/ undefined

```

Comment:
I think the best option for now is to disallow assertions on type-only imports/re-exports. There is currently just no use for them, since they get dropped from emit and they don’t yet imply any semantics. If, later, import assertions change our type checking behavior, it would probably make sense to allow them at that point. Allowing them for no good reason now might make it difficult to give certain assertions semantic meaning later.